packages/api/tests/test-endpoints-granules.js

```diff
@@ -325,3 +329,105 @@ test('move a granule', async (t) => {
     });
   });
 });
+
+test('move a file and update metadata', async (t) => {
+  const bucket = process.env.internal;
+  process.env.bucket = bucket;
+  const buckets = {
+    protected: {
+      name: process.env.internal,
+      type: 'protected'
+    },
+    public: {
+      name: randomString(),
+      type: 'public'
+    }
+  };
+
+  process.env.distEndpoint = 'http://example.com/';
+  await aws.s3().putObject({ Bucket: bucket, Key: `${process.env.stackName}/workflows/buckets.json`, Body: JSON.stringify(buckets) }).promise();
+
+  await aws.s3().createBucket({ Bucket: buckets.public.name }).promise();
+  const newGranule = fakeGranuleFactory();
+  const metadata = fs.createReadStream('tests/data/meta.xml');
+
+  newGranule.files = [
+    {
+      bucket,
+      name: `${newGranule.granuleId}.txt`,
+      filepath: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.txt`
+    },
+    {
+      bucket: buckets.public.name,
+      name: `${newGranule.granuleId}.cmr.xml`,
+      filepath: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.cmr.xml`,
+      filename: `s3://${buckets.public.name}/${process.env.stackName}/original_filepath/${newGranule.granuleId}.cmr.xml`
+    }
+  ];
+
+  await g.create(newGranule);
+
+  await Promise.all(newGranule.files.map(async (file) => {
+    if (file.name === `${newGranule.granuleId}.txt`) {
+      aws.s3().putObject({ Bucket: file.bucket, Key: file.filepath, Body: 'test data' }).promise();
+    }
+    else {
+      aws.s3().putObject({ Bucket: file.bucket, Key: file.filepath, Body: metadata }).promise();
+    }
+  }));
+
+  const destinationFilepath = `${process.env.stackName}/moved_granules`;
+  const destinations = [
+    {
+      regex: '.*.txt$',
+      bucket,
+      filepath: destinationFilepath
+    }
+  ];
+
+  const moveEvent = {
+    httpMethod: 'PUT',
+    pathParameters: {
+      granuleName: newGranule.granuleId
+    },
+    body: JSON.stringify({
+      action: 'move',
+      destinations
+    })
+  };
+
+  await testEndpoint(granuleEndpoint, moveEvent, async (response) => {
```

Comment:
testEndpoint expects a regular `callback` function. it seems that you are passing it a promise (`async (response)`)

Reply:
But when I remove the async, all the awaits I have in the testEndpoint start complaining

Reply:
you have to handle them without the usual async/await. For example you can return the whole response with a promise and then handle the rest outside of the testEndpoint function:

```
const response = await testEndpoint(granuleEndpoint, moveEvent, (r) => r);
```